gpytorch/lazy/cat_lazy_tensor.py

```diff
@@ -364,15 +364,32 @@ def devices(self):
     def device_count(self):
         return len(set(self.devices))
 
-    def to(self, device_id):
+    def to(self, *args, **kwargs):
         """
-        returns a new CatLazyTensor with device_id as the output_device
+        Returns a new CatLazyTensor with device as the output_device
         Warning: this does not move the LazyTensors in this CatLazyTensor to
-        device_id
+        device
         """
-        new_kwargs = dict(self._kwargs)
-        new_kwargs["output_device"] = device_id
-        return self.__class__(*self._args, **new_kwargs)
+        dtype = kwargs.pop("dtype", None)
+        device = kwargs.pop("device", None)
+
+        if dtype is None:
+            dtype_list = [x for x in args if type(x) is torch.dtype]
+            if len(dtype_list) > 0:
+                dtype = dtype_list[0]
+
+        if device is None:
+            device_list = [x for x in args if type(x) is torch.device]
+            if len(device_list) > 0:
+                device = device_list[0]
```

Comment:
feels like you could make a little type helper for this and then call it here and below to dedup code